keep.tex

```diff
@@ -530,38 +531,188 @@ \subsection{Concerns with active attacks}
+  result, $r_1 = threshold(r_0, s_{0\rightarrow{t}})$ where
+  $s_{0\rightarrow{t}}$ is the minimal shares necessary the the group
+  to produce a signature. \\
+  \textbf{Iteration:} & Each block published on the chain will include
+  a signature from $G_{registered}$ of the random value $r_i$. As the
+  chain grows, the signing threshold groups will change based on
+  provider availability. If any group is non-responsive up to its
+  threshold $t$, the group is removed from $G_{registered}$. \\
+  \textbf{Failure:} & Each iteration is an opportunity for a
+  group to fail to generate a valid signature. If a group $G_i$ fails
+  to sign the last iteration's random value, $G_{i+1}$ will be used
+  instead. \\
+  \hline
+\end{tabular}
+\end{table*}
 
-\subsection{The result registry}
+This work relies on the idea of threshold secret sharing schemes---
+secret sharing schemes that retain confidentiality up to some
+threshold $t$ of honest actors.
+
+Threshold signatures are a related idea. A threshold signature is a
+signature across $n$ parties that requires some minimum $t$ actively
+participating to sign. It's a similar idea to "multi-sig" as deployed
+in cryptocurrencies today.
+
+Traditional multi-sig, however, requires a smart contract on the
+blockchain to validate each signature and release funds. Threshold
+signature schemes actually require a threshold $t$ to construct a
+signature at all, removing a layer of complexity and coordination
+between parties.
+
+The use of threshold signatures means a number of participating
+signers in a signing group can be unavailable, and the signature will
+still succeed in the presence of $t$ functioning signers. This
+provides some of our beacon's required resilience in the face of
+failing or misbehaving nodes.
+
+If threshold signatures sound familiar, it might be because they're a
+core functionality keeps provide. For example, a keep signing a
+Bitcoin transaction does so using threshold ECDSA.
+
+A threshold relay is a way to chain threshold signatures to create a
+random beacon. Participants in a threshold relay form threshold
+groups. These groups generate new public keys that identify the group
+and correspond to a newly generated threshold private key, split
+across the participants.
+
+As providers join the network, they will form threshold groups. These
+groups will then sign a piece of random data from a trusted party,
+bootstrapping the relay. The resulting signature provides the random
+data for the next iteration, which can be verified by the rest of the
+network participants and rejected if invalid. Each iteration, a new
+signing group is chosen by the previous iteration's random value. As
+all groups sign the previous iteration's value, if a signature that's
+chosen is invalid, the signature from the next group in line can be
+chosen instead.
+
+\subsubsection{Keep selection group}
+
+Our threshold relay system will be composed of keep providers seeking
```

Comment:
I know it was addressed that this scheme should not have much of a performance impact, but what is the performance impact on top of sMPC? Should it be addressed or does this satisfy the concern?

Reply:
If anything this will be an improvement over past ideas for keep creation, especially from a performance and resilience perspective.

We'll implement this in two pieces- a light client that merely participates in the RNG (tiny resource usage) and delegates work, and sMPC node clients that do the hard work.

Reply:
Still trying to process all of this, isn't the secret (private key) sharing method in the threshold relay still sMPC?

Reply:
Threshold relay is a lighter, more specific application of similar ideas. There aren't any multiplication rounds, for example. Participants are generating everything mostly in parallel